webviz_subsurface/plugins/_reservoir_simulation_timeseries_regional.py

```diff
@@ -0,0 +1,1305 @@
+                                        {"label": i.lower().capitalize(), "value": i}
+                                        for i in [
+                                            "table",
+                                            "box plot",
+                                            "histogram",
+                                            "per realization",
+                                        ]
+                                    ],
+                                    value="table",
+                                    clearable=False,
+                                ),
+                            ],
+                        ),
+                        html.Span(children="Filters:", style={"font-weight": "bold"},),
+                        html.Div(id=self.uuid("filters"), children=self.selector_divs,),
+                    ],
+                ),
+                html.Div(
+                    style={"flex": 6},
+                    children=[
+                        html.Div(
+                            children=wcc.Graph(
+                                id=self.uuid("graph"),
+                                clickData={
+                                    "points": [{"x": str(self.smry["DATE"].min())}]
+                                },
+                                style={"height": "450px",},
+                            ),
+                        ),
+                        html.Div(
+                            id=self.uuid("stats_title"),
+                            style={"textAlign": "center",},
+                            children="",
+                        ),
+                        html.Div(id=self.uuid("date_view_wrapper")),
+                    ],
+                ),
+                dcc.Store(
+                    id=self.uuid("date"), data=json.dumps(str(self.smry["DATE"].min()))
+                ),
+                dcc.Store(id=self.uuid("ref_vec"), data=json.dumps("")),
+            ],
+        )
+
+    @property
+    def _callback_inputs(self):
+        """Returns all Dash inputs for selecting and filtering volumes
+        The number of inputs will vary depending on the available
+        selector columns in the volumes dataframe
+        """
+        return [
+            Input(self.selectors_id(selector), "value")
+            for selector in (
+                [
+                    "fip",
+                    "ensemble",
+                    "groupby",
+                    "vec",
+                    "timeseries_visualization",
+                    "date_view",
+                ]
+                + self.selectors
+            )
+        ]
+
+    def set_callbacks(self, app):
+        @app.callback(
+            [
+                Output(self.uuid("graph"), "figure"),
+                Output(self.uuid("date_view_wrapper"), "children"),
+                Output(self.uuid("ref_vec"), "data"),
+            ],
+            self._callback_inputs + [Input(self.uuid("date"), "data")],
```

Comment:
This might be a good opportunity to look at pattern matched callbacks https://dash.plotly.com/pattern-matching-callbacks

The select id's could be replaced by a dictionary and then pattern matching can be used in the callback.
```python
    def selectors_id(self, selected):
        return {
            "page": self.uuid(f"select"),
            "type": "select",
            "value": selected,
        }

def set_callbacks(self, app):
        @app.callback(
            [
                Output(self.uuid("graph"), "figure"),
                Output(self.uuid("date_view_wrapper"), "children"),
                Output(self.uuid("ref_vec"), "data"),
            ],
            [
                Input(
                    {
                        "page": self.uuid(f"select"),
                        "type": "select",
                        "value": ALL,
                    }, 'value'
                )
            ]
            + [Input(self.uuid("date"), "data")]
            
        )
        def _render_charts(*args):  # pylint: disable=too-many-locals
            inputs = dash.callback_context.inputs
            #...
```

Reply:
@HansKallekleiv  yes it could be a good idea. 
I've tested a bit more, and I think we have to make individual sets of selectors/filters for each FIP array selection, as they don't necessarilly have the same filters (e.g. not all are geolocial zone/region, but maybe FIPOWC is above/below initial contact and etc). Could then maybe set the type to the value of the FIP array selector, so that only the relevant filters are active. Should ideally hide the inactive ones as well. In theory we could probably with the pattern matching callbacks even create new filters when switching FIP array instead of hiding/unhiding?

Reply:
yes should be possible :)